extensions/emmet/src/abbreviationActions.ts

```diff
@@ -435,17 +442,32 @@ function expandAbbr(input: ExpandAbbreviationInput): string | undefined {
 
 	try {
 		// Expand the abbreviation
-		let expandedText = helper.expandAbbreviation(input.abbreviation, expandOptions);
+		let expandedText;
 
 		if (input.textToWrap) {
+			let parsedAbbr = helper.parseAbbreviation(input.abbreviation, expandOptions);
+			if (input.rangeToReplace.isSingleLine) {
+
+				// Fetch innermost element in the expanded abbreviation
+				let lastNode = parsedAbbr;
+				let tagName = parsedAbbr.name || '';
+				while (lastNode.children && lastNode.children.length > 0) {
```

Comment:
what if `parsedAbbr` was null or undefined?

Reply:
It'll throw an error when trying to obtain its property 'name', and we'll catch that. Do you think we should treat it differently?